File: go/app/acc-serv/internal/api/conf.go
//
// weproxy@foxmail.com 2022/10/29
//

package api

import (
	"io"

	"weproxy/acc/libgo/logx"
)

// newConfCli ...
func newConfCli() (io.Closer, error) {
	m := &confCli{}
	if err := m.Start(); err != nil {
		return nil, err
	}
	return m, nil
}

// confCli ...
type confCli struct {
}

// Start ...
func (m *confCli) Start() error {
	logx.D("%s confCli.Start()", TAG)
	return nil
}

// Close ...
func (m *confCli) Close() error {
	logx.D("%s confCli.Close()", TAG)
	return nil
}

File: cc/app/acc/internal/api/conf.cc
//
// weproxy@foxmail.com 2022/10/03
//

#include "api.h"
#include "gx/io/io.h"
#include "logx/logx.h"

namespace internal {
namespace api {

// confCli ...
struct confCli : public io::xx::closer_t {
    // Start ...
    error Start() {
        LOGF_D("%s confCli.Start()", TAG);
        return nil;
    }

    // Close ...
    error Close() {
        LOGF_D("%s confCli.Close()", TAG);
        return nil;
    }
};

// newConfCli ...
R<io::Closer, error> newConfCli() {
    auto m = NewRef<confCli>();
    auto err = m->Start();
    if (err != nil) {
        return {nil, err};
    }
    return {m, nil};
}

}  // namespace api
}  // namespace internal
